backend/app/verification.py

```diff
@@ -134,6 +134,11 @@ def _get_domain_name(app_id: str) -> str | None:
         projectname = _demangle_name(projectname)
         # https://sourceforge.net/p/forge/documentation/Project%20Web%20Services/
         return f"{projectname}.{domain}.io".lower()
+    elif len(app_id.split(".")) == 4:
+        [tld, domain, subdomain] = app_id.split(".")[0:3]
+        subdomain = _demangle_name(subdomain)
+        domain = _demangle_name(domain)
+        return f"{subdomain}.{domain}.{tld}".lower()
```

Comment:
the split should be done in reverse. In flathub we have strict policy of the last component being the app name, the rest being a FQDN in reverse.

Many countries have sub domain in the tld (`.jp`, `.uk`, `.in`, etc.) that are the top level for registration leading to at least 3 components for a domain name.

I don't see why you have to split at 3.

Reply:
I don't remember the details, but I think my hope was, to separate the subdomain issue from the tld issue and thus maybe make it easier to get a first change in.

Reply:
this https://github.com/flathub-infra/website/pull/3265 looks like the better approach.

Reply:
This is only the "better" version cause it has tests

Reply:
test on a broken feature is still a broken feature.

Reply:
not sure that's news